File: zig/src/render-console/consoleHighlightRenderer.zig
const std = @import("std");

const MarkdownNode = @import("../types/MarkdownNode.zig").MarkdownNode;
const ConsoleRendererState = @import("../types/RendererState.zig").RendererState;
const Renderer = @import("../types/Renderer.zig").Renderer;
const ansiBlack = @import("console.zig").ansiBlack;
const ansiYellowBg = @import("console.zig").ansiYellowBg;
const ansiReset = @import("console.zig").ansiReset;
const renderChildrenConsole = @import("console.zig").renderChildrenConsole;

pub const consoleHighlightRenderer = Renderer{
    .name = "highlight",
    .render = render,
};

pub fn render(node: *const MarkdownNode, state: *ConsoleRendererState, first: ?bool, last: ?bool, decode: ?bool) void {
    _ = first;
    _ = last;
    _ = decode;

    const style = ansiYellowBg ++ ansiBlack;

    state.output.appendSlice(state.allocator, style) catch unreachable;
    renderChildrenConsole(node, state, true) catch unreachable;
    state.output.appendSlice(state.allocator, ansiReset) catch unreachable;
}

File: dotnet/Allmark/RenderConsole/ConsoleHighlightRenderer.cs
namespace Allmark.Render;

using Allmark.Types;

public static class ConsoleHighlightRenderer
{
    public static OutputRenderer Create()
    {
        return new OutputRenderer
        {
            Name = "highlight",
            Render = (node, state, first, last, decode) => Render(node, state),
        };
    }

    public static void Render(MarkdownNode node, RendererState state)
    {
        var style = Ansi.YellowBg + Ansi.Black;
        state.Output.Append(style);
        RenderChildren.Execute(node, state);
        state.Output.Append(Ansi.Reset);
    }
}
